tests/conftest.py

```diff
@@ -77,3 +80,23 @@ def incomplete_application():
         factory = getattr(application_factories, factory_name)
         factory(application=application)
     return application
+
+
+slack_token = os.environ.get("SLACK_BOT_TOKEN")
+slack_test_channel = os.environ.get("SLACK_TEST_CHANNEL")
+
+
+@pytest.fixture
+def slack_messages(monkeypatch, enable_network):
+    messages = []
+
+    actual_post = services.slack.post
+
+    def post(text, channel):
+        messages.append((text, channel))
+
+        if slack_token and slack_test_channel:  # pragma: no cover
+            actual_post(text, slack_test_channel)
```

Comment:
Did you see if pytest-network would let us enable the network for just this block? (assumption: this is what `enable_network` was added for?)

Reply:
yep, although the pytest-network pypi README is wrong:
https://pypi.org/project/pytest-network/ vs
https://github.com/best-doctor/pytest_network